auth/clerk_client.go

```diff
@@ -154,34 +115,34 @@ func (h ClerkHandler) Session(next echo.HandlerFunc) echo.HandlerFunc {
 	}
 }
 
-func (h *ClerkHandler) getUserFromAccessToken(ctx context.Context, accessToken *models.AccessToken) (*models.Person, string, error) {
+func (h *ClerkHandler) getUserFromAccessToken(ctx context.Context, accessToken *models.AccessToken) AuthResult {
 	sessionID := accessToken.ID.String()
 	if user, exists := h.userCache.Get(sessionID); exists {
-		return user.(*models.Person), sessionID, nil
+		return AuthResult{User: user.(*models.Person), SessionID: sessionID}
 	}
 
 	dbUser, err := db.GetUserByID(ctx, accessToken.PersonID.String())
 	if err != nil {
-		return nil, "", fmt.Errorf("error fetching user by id[%s]: %w", accessToken.PersonID, err)
+		return AuthResult{Error: fmt.Errorf("error fetching user by id[%s]: %w", accessToken.PersonID, err)}
```

Comment:
I think the function should probably  return (`AuthResult, error`)

Reply:
updated